src/colvarmodule.cpp

```diff
@@ -712,18 +707,40 @@ int colvarmodule::calc()
     error_code |= write_traj_files();
   }
 
-  // write restart files, if needed
+  // write restart files and similar data
   if (restart_out_freq && (cvm::step_relative() > 0) &&
       ((cvm::step_absolute() % restart_out_freq) == 0) ) {
+
     if (restart_out_name.size()) {
       // Write restart file, if different from main output
       error_code |= write_restart_file(restart_out_name);
     } else {
       error_code |= write_restart_file(output_prefix()+".colvars.state");
     }
-    write_output_files();
+
+    cvm::increase_depth();
+    for (std::vector<colvar *>::iterator cvi = colvars.begin();
+         cvi != colvars.end();
+         cvi++) {
+      // TODO remove this when corrFunc becomes a bias
+      error_code |= (*cvi)->write_output_files();
+    }
+    for (std::vector<colvarbias *>::iterator bi = biases.begin();
+         bi != biases.end();
+         bi++) {
+      error_code |= (*bi)->write_state_to_replicas();
+    }
+    cvm::decrease_depth();
+
+    if (cv_traj_os != NULL) {
+      // do not close, there may be another run command
+      error_code |= proxy->flush_output_stream(cv_traj_os);
+    }
   }
 
+  // write output files for biases (use custom frequencies if provided)
+  error_code |= write_output_files();
+
```

Comment:
Why was this code taken out of `cvm::write_output_files()` ?

Reply:
Fixed in the rebased version.